format_spec.md

```diff
@@ -0,0 +1,289 @@
+
+## Version Objects
+
+Version Objects contain fields relating to a certain version of a Project. Below are all the fields that a Version Object may contain.
+
+### `id`
+
+The ID of this version. There is no required format for how to format a version ID, other than that they CANNOT include spaces.
+
+---
+
+### `name`
+
+The human-readable name of this version. It is RECOMMENDED to follow Semantic Versioning, but it is not required. Spaces MAY be used in this name.
+
+---
+
+### `semver` (optional)
+
+An optional Semantic Versioning compatible version number.
+
+---
+
+### `releaseDate`
+
+The release date of this verison. It MUST be stored as an ISO-8601 conforming String.
+
+---
+
+### `changelog`
+
+The changelog of this version. This MAY be moved into a separate API call for hosts. The changelog MAY contain HTML, Markdown, or other forms of rich text.
+
+---
+
+### `minecraftVersions`
+
+This field MUST be an array listing compatible Minecraft versions for this Project version.
+
+---
+
+### `dependencies` (optional)
+
+This field MUST be an array containing information about dependencies that may or may not be required for this project. Each object in this array contains the following fields:
+
+##### `type`
+
+The type of dependency this is. Valid options are:
+
+- `asset` - For example, a required library mod
+- `framework` - For example, a required modloader
+
+##### `src`
+
+This field contains where this dependency can be downloaded. It contains the name of the host, in an all-lowercase string. Examples include: `curseforge`, `diluv` and `modrinth`. If the `type` field is set to `framework`, this field CANNOT exist;
+
+##### `id`
+
+The ID of the required dependency. If the dependency is a required asset and not a framework, then the ID of the asset stored at the host specified in the `src` field should be used. If the dependency is a required framework, then a framework from the list of standardized framework IDs should be used:
+
+- `fabric-loader`
+- `forge`
+- `liteloader`
+- `rift`
+
+##### `required`
+
+This field MUST be present on dependencies with the `asset` type. It CANNOT be present on dependencies with the `framework` type. This field is a boolean that indicates whether a dependecy is required in order for the Project requiring it to function properly.
+
+##### `version`
+
+This field MUST be either an array or String. If this field is an array, it MUST contain a list of compatible versions. If this field is a String, it MUST contain a Semantic Versioning comparison String. An example of a comparison string is `>= 25.0.219`
+
+---
+
+### `files`
+
+This field contains the downloadable files of a Version. It MUST be an array of File Objects.
+File Objects contain information on a downloadable file. Fields are as follows:
+
+##### `name`
+
+The full name of this file, including file extension.
```

Comment:
Windows and Unix special characters are prohibited. Unicode allowed. Windows device filenames are prohibited. Filename must not begin or end with a dot (this includes . and ..) or whitespace or be blank.

Or, we can say that anything is allowed here but the installer doesn't have to honour it.